packages/terminal/src/browser/terminal-frontend-module.ts

```diff
@@ -49,12 +49,33 @@ export default new ContainerModule(bind => {
     bind(WidgetFactory).toDynamicValue(ctx => ({
         id: TERMINAL_WIDGET_FACTORY_ID,
         createWidget: (options: TerminalWidgetOptions) => {
+            let counter = terminalNum;
+            let domId = options.id || 'terminal-' + counter;
+            let termWidget: TerminalWidget | undefined;
+            if (options.id) {
```

Comment:
Since 

```
let domId = options.id || 'terminal-' + counter
```

I think we can reduce the `ifelse` into a single block of code which simply uses

```ts
termWidget = ctx.container.get(TerminalFrontendContribution).getById(domId);
```

As it is logically equivalent.

Reply:
factory should never reuse a widget, but always return a new 

the solution does not look good. Please could you point me to the client? It seems that it is passing bogus arguments to the widget manager and get a new terminal because of it.